apps/hooks-docs/src/examples/useIntersectionObserver.tsx

```diff
@@ -1,11 +1,11 @@
 import { DemoContainer } from "$/components/DemoContainer"
 import { useIntersectionObserver } from "@kaioken-core/hooks"
-import { signal, useRef } from "kaioken"
+import { useSignal, useRef } from "kaioken"
 
 export const UseIntersectionObserverExample: Kaioken.FC = () => {
-  const isVisible = signal(false)
+  const isVisible = useSignal(false)
 
-  const ref = useRef<HTMLElement | null>(null)
```

Comment:
question: does removing the null here. mean that the user can't set the `.current` value themselves now? I remember we talked alot about that but It's slipped my mind abit of how exactly that works again. I tried looking it up on the docs but couldn't find much about that behaviour

Reply:
probably best to look at react's docs regarding ref types. but either way it's fine in this case as the user provides the ref, and it can be a `RefObject` or `MutableRefObject`

Reply:
for clarity, creating a ref via `useRef<number>(null)` creates a `RefObject`. This is typed as readonly and is expected to only be set by kaioken or something else outside of userspace. 
creating a ref via `useRef<number>(42)` or `useRef<number | null>(null)` both create a `MutableRefObject` for which the user won't be warned for mutations.